File: JavaScript/Scaler/03-Lexical-Scope-Closures/counter.js
function counter() {
    let count = 0;

    return function () {
        return ++count;
    }
}

const c1 = counter();
console.log(c1()); // 1
console.log(c1()); // 2
console.log(c1()); // 3
console.log(c1()); // 4

console.log("------------------------------");

const c2 = counter();
console.log(c2()); // 1
console.log(c2()); // 2
console.log(c2()); // 3
console.log(c2()); // 4

File: Java/src/AdvanceProgrammingConcepts/InterfaceAbstractClass04/Static/Counter.java
package AdvanceProgrammingConcepts.InterfaceAbstractClass04.Static;

public class Counter {

    static int count;
    int x = 20;

    Counter(){
        count++;
    }

    static void staticMethod(){
//        System.out.println(x);
        System.out.println(count);
    }

    void nonStaticMethod(){
        System.out.println(x);
        System.out.println(count);
    }
}
